pkg/ebpf/compiler/compiler.go

```diff
@@ -140,8 +141,13 @@ func NewEBPFCompiler(headerDirs []string, verbose bool) (*EBPFCompiler, error) {
 	} else {
 		dirs, err = kernel.FindHeaderDirs()
 		if err != nil {
```

Comment:
You should probably only try to download headers if the dirs are not already there?

Reply:
That's the case here: if the headers are already on the machine, then `FindHeaderDirs` will return the header directories & `err` will be `nil`. If it isn't, then we didn't find the headers, so we'll try downloading them

Reply:
I thought there were other error cases as well, other than the headers not being found.